ImGui 1.92.+ support

mctb32 opened the pull request:
In regards to this issue: https://github.com/DiligentGraphics/DiligentTools/issues/249

- switched ImGui submodule to official repository version 1.92.1
- updated ImGuiDiligentRender to support the new texture API with scaling fonts 
- added public IMGUI_DEFINE_MATH_OPERATORS define to Diligent-Imgui target in CMakeLists.txt (not sure about this one, if someone wants to provide custom math operators, but the alternative was to add this define to every sample/tutorial project in diligent). 

I tested everything on Windows in all backends and it works nice. Other platforms should work as well as the changes were relatively straightforward, but some testing would of course be needed. 

I noticed that some Diligent samples use invalid ImGui flags for some controls and this is now being caught by assertions inside ImGui in debug builds. (for example ImGui::InputFloat() does not support ImGuiInputTextFlags_EnterReturnsTrue flag). These should be fixed, but they are in another repo.

TheMostDiligent commented:
The main problem was with the MacOS - did you have a chance to try it?

mctb32 commented:
> The main problem was with the MacOS - did you have a chance to test it?

Not yet. I'm new to Mac development but I'll try to test it.

mctb32 commented:
@TheMostDiligent Yes, I can see this happening for me too. Thanks for the info and sorry I missed that. 
I'll take a look at it over the weekend and see if I can figure something out.

TheMostDiligent commented:
I created a branch for imgui update.
Let's first merge the renderer changes, since they look good, and then work on MacOS issues to separate the problems.

mctb32 commented:
> I created a branch for imgui update. Let's first merge the renderer changes, since they look good, and then work on MacOS issues to separate the problems.

Great! Implemented the changes you requested.

TheMostDiligent commented:
Can you remove all macos changes from this PR - we will handle MacOS separately

mctb32 commented:
Yes - no problem, but if I do that, the MacOS build will be failing. Is that ok? 
It's failing in the old 1.85 macos backend that uses apis no longer available in the new imgui.

TheMostDiligent commented:
We can make some temporary fixes to make the files compile. We will fix them later. This is a work branch

mctb32 commented:
@TheMostDiligent  Ok. I reverted the macos related changes. But then I also updated the legacy 1.85 backend to use the new imgui IO api so it now compiles with 1.92.1+. I used some AI help + some bits from the new backends, but I also did some more extensive testing this time so I hope it works as expected.

TheMostDiligent commented:
Completed update to 1.92.1 in 5786b70f8078032200db904c156071c608030e44.
@mctb32 thank you for your help.

TheMostDiligent commented:
Yes. There were quite a lot of changes needed this time, hopefully it will be smoother from here.
